**Part 1:**  <span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">&nbsp;Write a query that finds the total number of sales for the year 2015. Your query cannot use any aggregates. All calculations should be done using the data/time functions in this week's learning. Your output should be in two columns-- year_sales and num_of_sales.</span>

In [7]:
SELECT 
    EXTRACT(YEAR FROM ord_date) AS year_sales, 
    COUNT(ord_id) AS num_of_sales
FROM 
    dsci_504.Orders
WHERE 
    EXTRACT(YEAR FROM ord_date) = 2015
GROUP BY 
    year_sales;

(1 row(s) affected)

Total execution time: 00:00:00.008

year_sales num_of_sales 2015.0 135

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Find the monthly sales for the same year. use the same output as above</span>

In [8]:
SELECT 
    TO_CHAR(ord_date, 'YYYY-MM') AS year_month,
    COUNT(*) AS num_of_sales
FROM 
    dsci_504.Orders
WHERE 
    EXTRACT(YEAR FROM ord_date) = 2015
GROUP BY
    TO_CHAR(ord_date, 'YYYY-MM')
ORDER BY 
    year_month;

(12 row(s) affected)

Total execution time: 00:00:00.005

year_month num_of_sales 2015-01 12 2015-02 11 2015-03 12 2015-04 23 2015-05 10 2015-06 8 2015-07 10 2015-08 9 2015-09 9 2015-10 14 2015-11 7 2015-12 10

Write a query to find the sales for all years in the data.

In [9]:
SELECT 
  EXTRACT(YEAR FROM ord_date) as sales_year,
  COUNT(ord_id) as num_of_sales,
  SUM(order_tot) as year_sales
FROM 
  dsci_504.Orders 
GROUP BY 
  sales_year
ORDER BY 
  sales_year;

(20 row(s) affected)

Total execution time: 00:00:00.010

sales_year num_of_sales year_sales 2000.0 150 728573.98 2001.0 137 696021.07 2002.0 123 565160.48 2003.0 112 523462.78 2004.0 119 577862.23 2005.0 131 635804.33 2006.0 154 754774.44 2007.0 124 598244.74 2008.0 122 607583.88 2009.0 129 651427.80 2010.0 138 655375.09 2011.0 131 630545.52 2012.0 126 610428.02 2013.0 132 637818.16 2014.0 113 534454.30 2015.0 135 652318.08 2016.0 133 602895.41 2017.0 134 641076.76 2018.0 135 642704.33 2019.0 121 558094.48

- How many sales were there in 2010?

In [10]:
SELECT 
    EXTRACT(YEAR FROM ord_date) AS year,
    COUNT(*) AS num_of_sales,
    SUM(order_tot) AS year_sales,
    COUNT(*) FILTER (WHERE EXTRACT(YEAR FROM ord_date) = 2010) AS sales_in_2010
FROM 
    dsci_504.Orders 
GROUP BY 
    EXTRACT(YEAR FROM ord_date);

(20 row(s) affected)

Total execution time: 00:00:00.015

year num_of_sales year_sales sales_in_2010 2002.0 123 565160.48 0 2019.0 121 558094.48 0 2016.0 133 602895.41 0 2017.0 134 641076.76 0 2009.0 129 651427.80 0 2007.0 124 598244.74 0 2008.0 122 607583.88 0 2005.0 131 635804.33 0 2006.0 154 754774.44 0 2018.0 135 642704.33 0 2014.0 113 534454.30 0 2010.0 138 655375.09 138 2013.0 132 637818.16 0 2015.0 135 652318.08 0 2001.0 137 696021.07 0 2003.0 112 523462.78 0 2011.0 131 630545.52 0 2004.0 119 577862.23 0 2000.0 150 728573.98 0 2012.0 126 610428.02 0

- How many sales were there in 2010? is 138 sales in 2010

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Write a query to track customer acquisition in 2018. Display the number of customers acquired each month. Your output should be limited to two columns. You can name the columns as you see fit.</span>

In [11]:
SELECT 
    EXTRACT(MONTH FROM cus_join_date) AS month_acquired, 
    COUNT(cus_id) AS num_of_customers
FROM 
    dsci_504.Customers
WHERE 
    EXTRACT(YEAR FROM cus_join_date) = 2018
GROUP BY 
    month_acquired;

(12 row(s) affected)

Total execution time: 00:00:00.009

month_acquired num_of_customers 1.0 7 2.0 10 3.0 11 4.0 8 5.0 13 6.0 8 7.0 6 8.0 14 9.0 6 10.0 9 11.0 17 12.0 13

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Write the same query to track customer acquisition over the entire dataset. You can customize your output to your liking</span>

In [12]:
SELECT 
    DATE_TRUNC('YEAR', cus_join_date) AS year_acquired,
    COUNT(cus_id) AS num_of_customers_acquired
FROM 
    dsci_504.Customers
GROUP BY 
    year_acquired;

(20 row(s) affected)

Total execution time: 00:00:00.015

year_acquired num_of_customers_acquired 2005-01-01T00:00:00-05:00 151 2017-01-01T00:00:00-05:00 155 2020-01-01T00:00:00-05:00 21 2013-01-01T00:00:00-05:00 132 2006-01-01T00:00:00-05:00 146 2010-01-01T00:00:00-05:00 131 2004-01-01T00:00:00-05:00 124 2009-01-01T00:00:00-05:00 149 2015-01-01T00:00:00-05:00 132 2011-01-01T00:00:00-05:00 145 2001-01-01T00:00:00-05:00 137 2002-01-01T00:00:00-05:00 122 2018-01-01T00:00:00-05:00 122 2003-01-01T00:00:00-05:00 153 2019-01-01T00:00:00-05:00 140 2008-01-01T00:00:00-05:00 127 2012-01-01T00:00:00-05:00 135 2014-01-01T00:00:00-05:00 119 2007-01-01T00:00:00-05:00 144 2016-01-01T00:00:00-05:00 134

Which year had the most customers? 2017-01-01T00:00:00-05:00 num_of_customers_acquired 155

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Write a query to find the day of week the customer with id 1800 joined. Output the customer's table ID, customer ID, lastname and day of the week they joined.</span>

In [13]:
SELECT 
  cus_id, 
  cus_last_name, 
  TO_CHAR(cus_join_date, 'Day') AS join_day 
FROM dsci_504.Customers 
WHERE 
  cus_id = 1800;

(1 row(s) affected)

Total execution time: 00:00:00.001

cus_id cus_last_name join_day 1800 Dickerson Thursday

- what is the last name of customer 1800 and dow of the week customer 1800 joined? the last name is (Dickerson) and  joined on Thursday

**Part 2:** <span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">&nbsp;Determine the number of days between a customer joining and the delivery of their order. Limit your query to the year 2018</span>

In [14]:
SELECT 
    (ord_date - cus_join_date) AS days_between_join_and_delivery
FROM 
    dsci_504.Orders 
JOIN 
    dsci_504.Customers 
ON 
    Orders.cus_id = Customers.cus_id
WHERE 
    EXTRACT(YEAR FROM cus_join_date) = 2018;

(122 row(s) affected)

Total execution time: 00:00:00.008

days_between_join_and_delivery -6204 -2243 -1152 -3569 -3400 -3906 261 173 -6043 -1861 -3792 -15 -4021 -5290 -3575 -5833 -6734 -355 -4008 -1499 -4302 -2242 -1797 -4110 -1012 -1170 -597 -1344 -5251 -6700 -4610 -1985 -5474 -1595 -6629 -6509 -4744 -5283 -4171 -1976 -1546 -6309 -3295 -3556 -6655 -5569 37 -5312 -1987 -3224 -5724 -3015 -3283 -4682 -3438 -5348 -4579 -140 -1149 225 -3666 -2874 -5733 -2811 -4236 -3751 416 -583 -3739 -576 -5460 -2091 -1756 -6068 400 6 -912 -2672 -4576 -6802 -5042 -3442 -3016 -5117 -2542 -5464 -3178 -683 -6117 -339 -602 -4669 -314 -20 -1232 -5493 -1938 -2680 -5977 -2113 -5634 -2379 444 225 -4104 -2844 -1604 -1179 -1990 -4468 -5346 -1332 -2113 -899 -3839 -6342 -4203 -6198 -4599 -127 -374 -630

- What was the shortest and longest delay between acquisition and delivery in 2016?

In [15]:
SELECT Orders.cus_id, MIN(ord_ship_date - cus_join_date) AS shortest_delay,
 MAX(ord_ship_date - cus_join_date) AS longest_delay
FROM dsci_504.Orders
JOIN dsci_504.Customers ON Customers.cus_id = Orders.cus_id
WHERE EXTRACT(YEAR FROM cus_join_date) = 2016 AND EXTRACT(YEAR FROM ord_ship_date) = 2016
GROUP BY orders.cus_id;

(4 row(s) affected)

Total execution time: 00:00:00.015

cus_id shortest_delay longest_delay 894 -106 -106 901 84 84 929 -95 -95 2251 -204 -204

**Part 3:** <span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">&nbsp;Perform a query to find all customers with orders fulfilled through the Dallas warehouse. This should be a simple query with no joins. Order your results by the total value of each customer's order.</span>

In [16]:
SELECT cus_id, SUM(order_tot) as total_order_value 
FROM dsci_504.Orders, dsci_504.Warehouses 
where Warehouses.warehouse_id = Orders.warehouse_id 
and warehouse_name = 'Dallas' 
GROUP BY Orders.cus_id 
ORDER BY total_order_value DESC;

(859 row(s) affected)

Total execution time: 00:00:00.013

cus_id total_order_value 1967 8580 852 8520 2246 8480 709 8460 1358 8460 1163 8448 125 8448 533 8388 1763 8344 1664 8320 1925 8238.99 994 8238.99 593 8161.99 2176 8161.99 925 8142.74 731 8131.19 1656 8123.49 854 8114.17 576 8085.8 2290 8065.74 2544 8031.09 406 8019.6 47 8000.69 484 8000.69 1645 8000 2106 8000 1294 8000 2521 8000 2149 8000 188 7981.77 371 7935.43 426 7932.59 976 7925.03 2577 7916.93 1574 7902.33 1851 7890.98 2524 7879.94 987 7879.94 1653 7879.94 752 7879.94 2131 7868.84 1736 7868.29 1191 7868.29 1961 7842.94 1317 7842.94 2023 7842.94 208 7842.94 322 7824.44 403 7813.34 464 7813.34 681 7813.34 138 7813.34 745 7813.34 2347 7805.95 1324 7805.95 1785 7805.95 1985 7750.45 1491 7728.26 610 7728.26 1528 7728.26 1257 7728.26 1929 7717.16 1238 7699.99 2530 7694.96 530 7694.96 417 7399 1350 7399 1997 7399 1256 7399 354 7399 2401 7399 1678 6949.79 1308 6933.59 328 6901.19 1314 6891.47 2117 6868.79 2039 6868.79 1042 6858.64 634 6842.65 902 6834.66 70 6794.27 2340 6794.27 94 6787.79 818 6778.7 1253 6778.7 768 6768.35 171 6693.46 2542 6693.46 1438 6693.46 1727 6679.58 1770 6650.8 2037 6590.49 1571 6584.24 757 6584.24 807 6543.68 802 6537.44 1951 6490.63 686 6395 1728 6395 2438 6395 162 6395 1564 6326.68 823 6282.44 539 6282.44 1286 6273.59 571 6252.94 1870 6252.94 2533 6252.94 687 6252.94 69 6252.94 866 6252.94 1675 6252.94 1634 6240.99 2463 6229.34 1993 6223.45 1428 6223.45 848 6223.45 1605 6223.45 1325 6223.45 2091 6206 1340 6206 1118 6204.93 1979 6179.2 25 6177 985 6177 1747 6175.94 710 6175.94 538 6167.24 2201 6161.51 2587 6152.66 1455 6148 1501 6148 1339 6148 1364 6146.94 1115 6146.94 1896 6146.94 1900 6146.94 1746 6134.96 2510 6134.96 471 6132.44 614 6124.8 1518 6081.3 264 6080.25 1102 6032 2095 6032 361 6032 2284 6030.96 789 6030.96 2568 6030.96 1155 6024.08 677 6002.69 214 5995.79 939 5995.68 1801 5974.64 1094 5974.64 1863 5974.64 1820 5974.64 796 5974.64 1304 5946.59 504 5932.56 938 5899 1717 5899 489 5899 1523 5899 2237 5899 981 5899 2437 5899 1377 5899 1068 5898.75 2385 5898.75 1868 5885 1411 5882.67 1510 5882.07 615 5857.5 2351 5857.5 54 5856.44 734 5856.44 1588 5851.22 1531 5849.25 1504 5848.19 227 5830 1236 5828.94 872 5801.45 194 5800 658 5800 2179 5799 1485 5765.75 2118 5765.7 108 5752.31 2092 5745.59 2581 5745.59 804 5745.59 2387 5735.46 1910 5735.46 1901 5725.43 1487 5725.43 2053 5725.43 547 5720 678 5720 2260 5720 1338 5718.96 2475 5718.96 2169 5717.37 1079 5717.37 521 5717.37 1536 5698.55 1096 5698.55 1663 5698.55 588 5698.55 835 5679.85 990 5677.05 1818 5671.67 563 5671.67 514 5656.41 1329 5656.41 149 5654.21 2076 5641.03 1346 5636.73 2239 5632.19 1966 5615.22 317 5615.22 1904 5615.22 2202 5613.65 2211 5613.65 1434 5613.65 329 5613.65 1204 5609.99 2374 5609.99 289 5600.41 1905 5592.47 2217 5592.47 261 5591.03 1659 5591.03 2155 5591.03 1558 5588.31 442 5588.31 96 5588.31 1815 5575.13 1598 5567.22 623 5536.94 2509 5510.94 528 5510.94 2258 5510.94 2085 5507.74 1216 5499 737 5499 1874 5499 51 5498.69 427 5497.94 1342 5482.87 350 5451.15 1524 5430.44 1826 5430.44 2229 5430.36 2044 5430.36 967 5422.56 2425 5406.96 1835 5406.96 1577 5406.96 233 5404.94 2278 5404.94 1412 5404.94 1842 5404.94 344 5379.45 2435 5375.99 2222 5375.99 2220 5375.99 182 5375.99 2307 5362.5 1950 5362.5 2400 5362.5 954 5350 1010 5350 1306 5343.75 2593 5341.2 1881 5317.5 115 5302.96 1038 5300 2573 5300 1203 5300 161 5300 2049 5295.9 1924 5275 1213 5271.99 1534 5271.99 2223 5237.5 857 5222.5 1552 5200 2363 5200 1232 5199 319 5199 2157 5170.56 1922 5146.29 150 5146.29 34 5099 1886 5099 2082 5085.6 1229 5083.56 1980 5077.15 2007 5063.75 112 5049.19 296 5046.31 869 5040.75 1212 5040.75 2250 5036.05 860 5036.05 2574 5036.05 2081 5029.7 2164 5029.7 1591 5029.7 2515 5029 449 5029 1392 5024.17 413 5023.13 2127 5018.23 1671 5017.84 2058 5017.05 2234 5006.53 1949 5005.98 2102 5005.5 335 5000 2421 4983.14 1839 4982.82 515 4982 1543 4982 374 4982 2473 4981.41 42 4976.66 1612 4975.13 652 4971.21 1692 4965.4 258 4963.2 1234 4959.74 2460 4959.74 1749 495

**Part 4**: Shimano has reached out to OPC and has asked that OPC provide a listing of all customers that purchased a bike with a combination of Shimano's race components XTR due to a critical recall issuance. They have concerns that not all purchasers have registered their bikes or that they may be out of warranty. Write a query that outputs the following elements to a table that can be exported and provided to Shimano. The table should be stored in the public schema.

- Customer ID
- Customer Last Name
- Customer First Name
- Customer State
- Customer Phone Number

In [17]:
SELECT 
    c.cus_id, 
    c.cus_last_name, 
    c.cus_first_name, 
    c.cus_state, 
    c.cus_phone,
    comp.comp_name
FROM 
    dsci_504.Customers c
join dsci_504.orders o    
on c.cus_id = o.cus_id    
join dsci_504.builds b
on o.build_id = b.build_id
join dsci_504.components comp
on comp.comp_name like 'Shimano XTR%' 

(12995 row(s) affected)

Displaying Top 5000 rows.

Total execution time: 00:00:00.298

cus_id cus_last_name cus_first_name cus_state cus_phone comp_name 27 Jones Noel 6 5225885073 Shimano XTR M9100 27 Jones Noel 6 5225885073 Shimano XTR 27 Jones Noel 6 5225885073 Shimano XTR M9100 27 Jones Noel 6 5225885073 Shimano XTR 27 Jones Noel 6 5225885073 Shimano XTR Trail 28 Garth Brandon 16 9014059049 Shimano XTR M9100 28 Garth Brandon 16 9014059049 Shimano XTR 28 Garth Brandon 16 9014059049 Shimano XTR M9100 28 Garth Brandon 16 9014059049 Shimano XTR 28 Garth Brandon 16 9014059049 Shimano XTR Trail 29 Saylor Farris 18 213240959 Shimano XTR M9100 29 Saylor Farris 18 213240959 Shimano XTR 29 Saylor Farris 18 213240959 Shimano XTR M9100 29 Saylor Farris 18 213240959 Shimano XTR 29 Saylor Farris 18 213240959 Shimano XTR Trail 30 Tompsen Taylor 26 3737170851 Shimano XTR M9100 30 Tompsen Taylor 26 3737170851 Shimano XTR 30 Tompsen Taylor 26 3737170851 Shimano XTR M9100 30 Tompsen Taylor 26 3737170851 Shimano XTR 30 Tompsen Taylor 26 3737170851 Shimano XTR Trail 31 Stevens Jonathon 25 2276951884 Shimano XTR M9100 31 Stevens Jonathon 25 2276951884 Shimano XTR 31 Stevens Jonathon 25 2276951884 Shimano XTR M9100 31 Stevens Jonathon 25 2276951884 Shimano XTR 31 Stevens Jonathon 25 2276951884 Shimano XTR Trail 32 Cooper Jack 30 7697046970 Shimano XTR M9100 32 Cooper Jack 30 7697046970 Shimano XTR 32 Cooper Jack 30 7697046970 Shimano XTR M9100 32 Cooper Jack 30 7697046970 Shimano XTR 32 Cooper Jack 30 7697046970 Shimano XTR Trail 33 Applewood Eric 4 5314070768 Shimano XTR M9100 33 Applewood Eric 4 5314070768 Shimano XTR 33 Applewood Eric 4 5314070768 Shimano XTR M9100 33 Applewood Eric 4 5314070768 Shimano XTR 33 Applewood Eric 4 5314070768 Shimano XTR Trail 34 Peters Jonathon 21 5640312585 Shimano XTR M9100 34 Peters Jonathon 21 5640312585 Shimano XTR 34 Peters Jonathon 21 5640312585 Shimano XTR M9100 34 Peters Jonathon 21 5640312585 Shimano XTR 34 Peters Jonathon 21 5640312585 Shimano XTR Trail 35 Manning Robert 7 7597449461 Shimano XTR M9100 35 Manning Robert 7 7597449461 Shimano XTR 35 Manning Robert 7 7597449461 Shimano XTR M9100 35 Manning Robert 7 7597449461 Shimano XTR 35 Manning Robert 7 7597449461 Shimano XTR Trail 36 Smith Brandt 23 495761977 Shimano XTR M9100 36 Smith Brandt 23 495761977 Shimano XTR 36 Smith Brandt 23 495761977 Shimano XTR M9100 36 Smith Brandt 23 495761977 Shimano XTR 36 Smith Brandt 23 495761977 Shimano XTR Trail 37 Delacruz Timothy 4 8440531484 Shimano XTR M9100 37 Delacruz Timothy 4 8440531484 Shimano XTR 37 Delacruz Timothy 4 8440531484 Shimano XTR M9100 37 Delacruz Timothy 4 8440531484 Shimano XTR 37 Delacruz Timothy 4 8440531484 Shimano XTR Trail 38 Jackson Harper 15 8759798525 Shimano XTR M9100 38 Jackson Harper 15 8759798525 Shimano XTR 38 Jackson Harper 15 8759798525 Shimano XTR M9100 38 Jackson Harper 15 8759798525 Shimano XTR 38 Jackson Harper 15 8759798525 Shimano XTR Trail 39 Sherman Mads 6 9772062009 Shimano XTR M9100 39 Sherman Mads 6 9772062009 Shimano XTR 39 Sherman Mads 6 9772062009 Shimano XTR M9100 39 Sherman Mads 6 9772062009 Shimano XTR 39 Sherman Mads 6 9772062009 Shimano XTR Trail 40 Dickerson Thomas 30 4870812999 Shimano XTR M9100 40 Dickerson Thomas 30 4870812999 Shimano XTR 40 Dickerson Thomas 30 4870812999 Shimano XTR M9100 40 Dickerson Thomas 30 4870812999 Shimano XTR 40 Dickerson Thomas 30 4870812999 Shimano XTR Trail 41 Robinson Albert 11 1951557552 Shimano XTR M9100 41 Robinson Albert 11 1951557552 Shimano XTR 41 Robinson Albert 11 1951557552 Shimano XTR M9100 41 Robinson Albert 11 1951557552 Shimano XTR 41 Robinson Albert 11 1951557552 Shimano XTR Trail 42 MacBain Eric 18 3560967852 Shimano XTR M9100 42 MacBain Eric 18 3560967852 Shimano XTR 42 MacBain Eric 18 3560967852 Shimano XTR M9100 42 MacBain Eric 18 3560967852 Shimano XTR 42 MacBain Eric 18 3560967852 Shimano XTR Trail 43 Riochardson Kenadi 15 2945999508 Shimano XTR M9100 43 Riochardson Kenadi 15 2945999508 Shimano XTR 43 Riochardson Kenadi 15 2945999508 Shimano XTR M9100 43 Riochardson Kenadi 15 2945999508 Shim

- What is the customer ID of the customer on line 20? 30

**Part 5:** Write a query that will aggregate the total of all bikes sold. Group the output by the bike manufacturer. Your output should contain the following elements:

- Bike Manufacturer
- Bike Name
- Bike Description
- Bike Country of Origin
- Total of all sales

In [18]:
SELECT 
    p.prod_manufacturer AS "Bike Manufacturer", 
    p.prod_name AS "Bike Name", 
    p.prod_description AS "Bike Description", 
    p.country_origin AS "Bike Country of Origin", 
    SUM(o.order_tot) AS "Total of all sales"
FROM 
     dsci_504. Products p 
	INNER JOIN   dsci_504.Orders o ON p.prod_id = o.prod_id
GROUP BY 
    p.prod_manufacturer, p.prod_name, p.prod_description, p.country_origin;

(68 row(s) affected)

Total execution time: 00:00:00.011

Bike Manufacturer Bike Name Bike Description Bike Country of Origin Total of all sales 3 Rip9 Full-Suspension Mountain Bike 2,3 161450.3 64 Megatrail Full-Suspension Mountain Bike 2,3 207981.74 64 Trail Pistol Full-Suspension Mountain Bike 2,3 219938.50 15 Following Full-Suspension Mountain Bike 2,3,4 225652.19 10 T-Series Full-Suspension Mountain Bike 2,3 124997.62 17 Scalpel Full-Suspension Mountain Bike 1,2 131881.05 14 Thunderbolt Full-Suspension Mountain Bike 1 103154.90 18 Reactor Full-Suspension Mountain Bike 3,4 177031.42 67 Haste Full-Suspension Mountain Bike 2,3 180871.86 7 Define Full-Suspension Mountain Bike 2,3 130694.83 26 Enduro Full-Suspension Mountain Bike 2,3 193599.34 7 Snabb Full-Suspension Mountain Bike 2,3 101909.64 3 Air9 Hardtail Mountain Bike 2,3 81477.28 70 Troy Full-Suspension Mountain Bike 2 134170.08 25 SB150 Full-Suspension Mountain Bike 3,4 306540 22 Ripmo Full-Suspension Mountain Bike 2,3,4 155615.77 8 Process 153 Full-Suspension Mountain Bike 2,3,4 118982.00 18 Mega Full-Suspension Mountain Bike 2,3,4 179285.05 26 Stumpjumper Full-Suspension Mountain Bike 2,3 194885.09 3 Jet9 Full-Suspension Mountain Bike 2,3 132057.13 3 RKT9 Full-Suspension Mountain Bike 1,2,3 155181.6 15 Insurgent Full-Suspension Mountain Bike 1,2,3 199298.94 1 X Trance Full-Suspension Mountain Bike 2,3,4 212526.77 17 Habit Full-Suspension Mountain Bike 2,3 146300.44 22 Mojo Full-Suspension Mountain Bike 2,3 255649.16 63 Tallboy Full-Suspension Mountain Bike 2,3,4 92438.37 25 SB140 Full-Suspension Mountain Bike 2,3,4 188543.5 43 AMD Full-Suspension Mountain Bike 2,3 187747.24 66 Firebird Full-Suspension Mountain Bike 2,3,4 259038.39 25 SB160 Full-Suspension Mountain Bike 3,4 324004.05 20 Tracer Full-Suspension Mountain Bike 1,2 196193.77 63 Chameleon Hardtail Mountain Bike 1,2,3 71412.49 25 SB115 Full-Suspension Mountain Bike 1,2,3 197773.66 14 Element Full-Suspension Mountain Bike 1,2 172432.23 62 Meta Am Full-Suspension Mountain Bike 2,3,4 235024.36 10 G-Series Full-Suspension Mountain Bike 4 153678.99 64 Gnarvana Full-Suspension Mountain Bike 3,4 280900.42 61 Occam Full-Suspension Mountain Bike 2,3 319280.00 64 Shredd Dog Full-Suspension Mountain Bike 3,4 208804.85 63 Hightower Full-Suspension Mountain Bike 2,3,4 177261.72 1 Trance Full-Suspension Mountain Bike 2,3,4 116946.38 69 Sight Full-Suspension Mountain Bike 1,2,3 226544.40 25 SB130 Full-Suspension Mountain Bike 2,3,4 167994.75 67 Onset Full-Suspension Mountain Bike 2,3, 199980.00 68 SuperFoxy Full-Suspension Mountain Bike 3,4 230741.87 62 Clash Full-Suspension Mountain Bike 2,3,4 208997.77 43 Stereo Full-Suspension Mountain Bike 2,3,4 235657.41 8 Hei Hei Full-Suspension Mountain Bike 2,3 208890.22 68 Foxy Full-Suspension Mountain Bike 2,3 175908.75 20 Primer Full-Suspension Mountain Bike 2,3,4 127086.05 24 Sensor Full-Suspension Mountain Bike 2,3 124729.08 22 DV9 Hardtail Mountain Bike 1 113877.15 61 OIZ Full-Suspension Mountain Bike 1 142216.55 15 Wreckoning Full-Suspension Mountain Bike 2,3,4 278751.80 25 SB100 Full-Suspension Mountain Bike 1,2 123426.42 17 Jekyll Full-Suspension Mountain Bike 1,2,3 178030.41 24 Force Full-Suspension Mountain Bike 2,3,4 116708.40 10 E-Series Full-Suspension Mountain Bike 3,4 269617.46 20 Sniper Full-Suspension Mountain Bike 1,2 100955.76 63 Megatower Full-Suspension Mountain Bike 2,3,4 211932.95 65 Rail Full-Suspension Mountain Bike 2,3 348444.86 64 The Smash Full-Suspension Mountain Bike 3,4 158426.09 69 Optic Full-Suspension Mountain Bike 2,3 169617.95 14 Altitude Full-Suspension Mountain Bike 2,3 313788.49 8 Process 134 Full-Suspension Mountain Bike 1,2,3 174591.74 18 Scout Hardtail Mountain Bike 2,3 99342.09 22 Ripley Full-Suspension Mountain Bike 2,3 192953.52 14 Instinct Full-Suspension Mountain Bike 2,3 192800.82

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Sort your output by the sum of the total order, manufacturer, and the bike name</span>

In [19]:
SELECT 
    Products.prod_manufacturer AS "Bike Manufacturer", 
    Products.prod_name AS "Bike Name",
    Products.prod_description AS "Bike Description",
    Products.country_origin AS "Bike Country of Origin",
    SUM(Orders.order_tot) AS "Total Sales"
FROM 
     dsci_504.Products
INNER JOIN 
     dsci_504.Orders ON Products.prod_id = Orders.prod_id
GROUP BY 
    Products.prod_manufacturer,
    Products.prod_name,
    Products.prod_description,
    Products.country_origin
ORDER BY 
    "Total Sales" DESC, 
    "Bike Manufacturer" ASC, 
    "Bike Name" ASC;

(68 row(s) affected)

Total execution time: 00:00:00.009

Bike Manufacturer Bike Name Bike Description Bike Country of Origin Total Sales 65 Rail Full-Suspension Mountain Bike 2,3 348444.86 25 SB160 Full-Suspension Mountain Bike 3,4 324004.05 61 Occam Full-Suspension Mountain Bike 2,3 319280.00 14 Altitude Full-Suspension Mountain Bike 2,3 313788.49 25 SB150 Full-Suspension Mountain Bike 3,4 306540 64 Gnarvana Full-Suspension Mountain Bike 3,4 280900.42 15 Wreckoning Full-Suspension Mountain Bike 2,3,4 278751.80 10 E-Series Full-Suspension Mountain Bike 3,4 269617.46 66 Firebird Full-Suspension Mountain Bike 2,3,4 259038.39 22 Mojo Full-Suspension Mountain Bike 2,3 255649.16 43 Stereo Full-Suspension Mountain Bike 2,3,4 235657.41 62 Meta Am Full-Suspension Mountain Bike 2,3,4 235024.36 68 SuperFoxy Full-Suspension Mountain Bike 3,4 230741.87 69 Sight Full-Suspension Mountain Bike 1,2,3 226544.40 15 Following Full-Suspension Mountain Bike 2,3,4 225652.19 64 Trail Pistol Full-Suspension Mountain Bike 2,3 219938.50 1 X Trance Full-Suspension Mountain Bike 2,3,4 212526.77 63 Megatower Full-Suspension Mountain Bike 2,3,4 211932.95 62 Clash Full-Suspension Mountain Bike 2,3,4 208997.77 8 Hei Hei Full-Suspension Mountain Bike 2,3 208890.22 64 Shredd Dog Full-Suspension Mountain Bike 3,4 208804.85 64 Megatrail Full-Suspension Mountain Bike 2,3 207981.74 67 Onset Full-Suspension Mountain Bike 2,3, 199980.00 15 Insurgent Full-Suspension Mountain Bike 1,2,3 199298.94 25 SB115 Full-Suspension Mountain Bike 1,2,3 197773.66 20 Tracer Full-Suspension Mountain Bike 1,2 196193.77 26 Stumpjumper Full-Suspension Mountain Bike 2,3 194885.09 26 Enduro Full-Suspension Mountain Bike 2,3 193599.34 22 Ripley Full-Suspension Mountain Bike 2,3 192953.52 14 Instinct Full-Suspension Mountain Bike 2,3 192800.82 25 SB140 Full-Suspension Mountain Bike 2,3,4 188543.5 43 AMD Full-Suspension Mountain Bike 2,3 187747.24 67 Haste Full-Suspension Mountain Bike 2,3 180871.86 18 Mega Full-Suspension Mountain Bike 2,3,4 179285.05 17 Jekyll Full-Suspension Mountain Bike 1,2,3 178030.41 63 Hightower Full-Suspension Mountain Bike 2,3,4 177261.72 18 Reactor Full-Suspension Mountain Bike 3,4 177031.42 68 Foxy Full-Suspension Mountain Bike 2,3 175908.75 8 Process 134 Full-Suspension Mountain Bike 1,2,3 174591.74 14 Element Full-Suspension Mountain Bike 1,2 172432.23 69 Optic Full-Suspension Mountain Bike 2,3 169617.95 25 SB130 Full-Suspension Mountain Bike 2,3,4 167994.75 3 Rip9 Full-Suspension Mountain Bike 2,3 161450.3 64 The Smash Full-Suspension Mountain Bike 3,4 158426.09 22 Ripmo Full-Suspension Mountain Bike 2,3,4 155615.77 3 RKT9 Full-Suspension Mountain Bike 1,2,3 155181.6 10 G-Series Full-Suspension Mountain Bike 4 153678.99 17 Habit Full-Suspension Mountain Bike 2,3 146300.44 61 OIZ Full-Suspension Mountain Bike 1 142216.55 70 Troy Full-Suspension Mountain Bike 2 134170.08 3 Jet9 Full-Suspension Mountain Bike 2,3 132057.13 17 Scalpel Full-Suspension Mountain Bike 1,2 131881.05 7 Define Full-Suspension Mountain Bike 2,3 130694.83 20 Primer Full-Suspension Mountain Bike 2,3,4 127086.05 10 T-Series Full-Suspension Mountain Bike 2,3 124997.62 24 Sensor Full-Suspension Mountain Bike 2,3 124729.08 25 SB100 Full-Suspension Mountain Bike 1,2 123426.42 8 Process 153 Full-Suspension Mountain Bike 2,3,4 118982.00 1 Trance Full-Suspension Mountain Bike 2,3,4 116946.38 24 Force Full-Suspension Mountain Bike 2,3,4 116708.40 22 DV9 Hardtail Mountain Bike 1 113877.15 14 Thunderbolt Full-Suspension Mountain Bike 1 103154.90 7 Snabb Full-Suspension Mountain Bike 2,3 101909.64 20 Sniper Full-Suspension Mountain Bike 1,2 100955.76 18 Scout Hardtail Mountain Bike 2,3 99342.09 63 Tallboy Full-Suspension Mountain Bike 2,3,4 92438.37 3 Air9 Hardtail Mountain Bike 2,3 81477.28 63 Chameleon Hardtail Mountain Bike 1,2,3 71412.49

- What was the bike with the most sales? Rail
- What was the total of sales for the Nukeproof Scout? 99342.09

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Write a query to find the total value of all bikes returned by customers. Your output should include the basic customer info, amount of return, and date of return. You can coordinate the output as you see fit but be sure to not include any PII data</span>

In [20]:
SELECT 
    cus.cus_first_name, 
    cus.cus_last_name, 
    cus.cus_join_date, 
    rac.tot_ret_amnt AS Return_Amount,
    rac.ord_id AS Date_Of_Return
FROM 
    dsci_504.Customers cus
INNER JOIN 
    dsci_504.Racs rac ON cus.cus_id = rac.cus_id;

(1024 row(s) affected)

Total execution time: 00:00:00.030

cus_first_name cus_last_name cus_join_date return_amount date_of_return Miguel Deleon 2001-08-10 0 5484681 Kristoff Dawson 2005-01-10 1008 6727420 Brandon Robinson 2008-05-23 0 4711571 Eric Rice 2006-02-13 970 3555795 Mike Smith 2019-01-05 0 5687001 Brian Acosta 2006-09-02 71 1985674 Nadine Slain 2005-03-01 0 5994079 Maggs Saylor 2006-03-24 0 5087861 Abigail Tankersly 2016-05-28 0 6077808 Erin Manning 2014-12-24 60 6241665 Robert Smith 2016-07-21 0 5677372 Mads Applewood 2006-11-14 1026 3695577 Noel Roberts 2006-05-15 1007 6047294 Eric Jackson 2019-11-15 49 3236261 Farris Applewood 2010-06-30 742 5868564 Maggs Aaronson 2014-09-19 0 6500666 Brice Thomason 2009-01-04 39 5255967 Brandon Smith 2019-12-13 615 4924882 Eric Robinson 2008-06-27 632 1791485 George Dewey 2010-06-28 0 6894826 Corey Presley 2005-10-26 0 3986347 Adam Shelton 2015-04-24 0 3560653 Joshua Roberts 2017-06-12 0 3326659 Margo Peters 2011-04-08 195 3779453 Noel Delacruz 2013-09-02 0 7311476 Reginald Deleon 2012-12-03 419 2177932 Eric Sanchez 2012-01-29 1041 5086249 Noel Harper 2002-06-06 203 4205221 Taylor Smith-Warren 2018-09-16 785 1455991 Joshua Cooper 2007-10-12 1079 3193964 Thomas Etzler 2006-12-29 0 6464145 Farris Connery 2011-12-31 0 5321869 Thomas Dawson 2016-02-07 392 5236088 Eric Shelton 2006-09-28 0 3681241 Carl Calahan 2019-07-14 0 4471131 Kenadi Jackson 2011-11-16 535 1929235 Mike Jackson 2012-08-07 642 1694460 Zachary Ellis 2003-11-12 425 6562501 Mads Applewood 2006-11-14 562 3303963 Farris Aaronson 2012-05-11 773 6755650 Noel Smith-Warren 2007-01-13 0 5649473 Blain Smith-Warren 2005-05-19 386 2629583 Alister Willis 2003-07-31 0 1458147 Corey Conley 2001-06-13 0 5168047 Leah Shelton 2006-05-02 0 5651788 Brice Riochardson 2011-03-24 0 2694750 Valerie Jones 2011-10-31 735 5861427 Margo Dickerson 2018-10-11 0 6556364 Adam Johnasen 2010-10-01 1066 2293928 Erin Dewey 2006-03-14 339 3272347 Moritz Smith 2011-05-30 176 5851299 Maggs Peters 2005-05-15 581 2845076 Mads Williamson 2012-12-19 0 5548388 Cooper Etzler 2007-06-03 0 4814298 Moritz Delapena 2016-12-13 69 2618832 Taylor Barrister 2008-12-28 0 5393236 Mike Jackson 2012-08-07 0 4827815 Mads Booker 2011-11-17 1018 7083064 Mike Maurer 2018-03-16 813 1955676 Brandon Archibald 2019-11-14 360 7304261 Reginald Peters 2005-09-07 807 1620794 Timothy Monk 2010-10-17 102 4344423 Albert Presten 2017-10-28 0 4366976 Miguel Garth 2004-02-03 0 2069011 Margo Jones 2017-03-12 0 5968744 Abigail Rowley 2009-04-18 505 6727185 Mindy Presten 2019-07-11 0 3984108 Valerie Connery 2002-11-10 391 3775746 Alister Hart 2006-08-03 596 3444796 Corey Garcia 2010-09-27 1024 7276203 Summer Smith-Warren 2003-12-10 581 2107831 Piper Maurer 2010-07-11 0 4460714 Manuel Stevens 2009-11-19 446 4206987 Reginald Williamson 2014-09-17 0 1332784 Jonathon Willis 2013-05-22 0 4205785 Erin Jordan 2007-10-21 767 2678429 Erin Jackson 2001-06-23 326 3758613 Erin Presley 2018-11-26 1092 5780157 Corey MacBain 2001-05-04 844 7186133 Cooper Smith-Warren 2014-03-12 1093 4083483 Lee Brooks 2007-06-17 657 5040689 Alister Saylor 2006-11-18 0 6419973 Aaron Dewey 2010-08-16 1010 3983170 Jack Jordan 2013-05-07 169 3320884 Harper Acosta 2018-06-05 0 6201595 George Garth 2019-06-29 0 3951017 Joshua Harper 2009-05-08 358 4195011 Aaron Roberts 2009-12-16 864 5046694 Thomas Ridgley 2007-10-17 0 2340380 Erin MacBain 2017-12-03 1049 1450563 Timothy Seiler 2001-12-26 0 2357460 Manuel Rice 2006-10-04 320 4433548 Mike Jackson 2012-08-07 711 3316575 Alister Riochardson 2007-01-24 0 1229905 Leah Smith 2002-01-31 756 6873054 Cooper Deleon 2004-05-03 381 6563406 Alister Thomason 2014-09-15 0 6106329 Brandon Acosta 2003-12-12 22 3513797 Nadine Monk 2005-04-16 0 7231583 Jack Saylor 2006-06-04 0 4464272 Manuel Booker 2017-08-05 85 4606107 Samuel Moore 2018-08-23 405 2986403 Jasmine Tompsen 2010-02-22 0 4690371 Eric Barrister 2018-01-25 568 2804411 Abigail Sanchez 2003-09-22 0 2099915 Abigail Slain 2005-08-18 122 4332112 Margo Dawson 2016-02-25 0 1588017 Jonathon Saylor 2003-1

- How many customers have returns? 1024
- What is the highest amount returned by a customer? 1845
- What is one unique thing about this query's results? This query returns the customer (without any PII  with the highest total return amount, the total value of their returns, and the id of their last return.

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Write a query to calculate the average bike price, the sum of all bikes sold, and the difference between the product price and sale price of every bike purchase. You can include any other columns you feel necessary</span>

In [21]:
SELECT
  AVG(P.prod_price) AS Average_Bike_Price,
  SUM(O.order_tot) AS Total_Sales,
  (P.prod_price - O.order_tot) AS Price_Difference,
  C.comp_name AS Component_Name
FROM 
  dsci_504.Products P 
  JOIN dsci_504.Orders O ON P.prod_id = O.prod_id
  JOIN dsci_504.Components C ON C.comp_id = P.prod_id
GROUP BY 
  C.comp_name,p.prod_price, o.order_tot;

(871 row(s) affected)

Total execution time: 00:00:00.041

average_bike_price total_sales price_difference component_name 7699.9900000000000000 16516.48 -558.25 Rocky Mountain AM 5000.0000000000000000 31200 -200 Cannondale 2 Flat 3999.0000000000000000 3418.93 580.07 Sram G2 R 5199.0000000000000000 5556.43 -357.43 Giant Connect R 4504.9900000000000000 14494.8 -326.61 Sram Code RSC 5271.9900000000000000 16923.09 -369.04 Evil Boomstick Carbon 3999.0000000000000000 8341.92 -171.96 GT Alloy 4200.0000000000000000 4403.7 -203.7 Cannondale 3 Riser 5295.9000000000000000 16761.51 -291.27 ProTaper Carb 3999.9900000000000000 16959.96 -240.00 Intense Recon Alloy 4227.9900000000000000 13286.46 -200.83 RaceFace Ride 4350.0000000000000000 17400 0 Nukeproof Horizon V2 3999.9900000000000000 16711.96 -178.00 Intense Recon Alloy 3500.0000000000000000 11261.25 -253.75 Rida Alpha R27 7699.9900000000000000 8031.09 -331.10 Rocky Mountain AM 3673.9900000000000000 3879.73 -205.74 Sram Code R 4299.9900000000000000 13802.97 -301.00 OC1 7699.9900000000000000 16131.48 -365.75 Rocky Mountain AM 4449.9000000000000000 9545.04 -322.62 ProTaper AL 4645.9900000000000000 4941.01 -295.02 Shimano SLX M7100 4299.9900000000000000 4573.04 -273.05 OC1 1850.0000000000000000 3875.76 -87.88 Shimano Deore M6100 2 3799.0000000000000000 4017.44 -218.44 RaceFace Ride 4679.0000000000000000 4880.2 -201.2 Magura MT7 5099.0000000000000000 10860.88 -331.44 Whyte 6061 Alloy 1949.9900000000000000 6083.97 -78.00 RaceFace NextR Carbon 5500.0000000000000000 5736.5 -236.5 OC3 Carbon 5375.9900000000000000 5631.35 -255.36 Shimano Deore M6100 2 4745.0000000000000000 15017.94 -260.98 Burgtec Alloy 4449.9000000000000000 9522.78 -311.49 ProTaper AL 3209.0800000000000000 9627.24 0.00 Sram G2 ULT 5899.0000000000000000 24917.36 -330.34 NS License Carb 3999.0000000000000000 17095.72 -274.93 Sram G2 R 7565.6600000000000000 32002.76 -435.03 Tektro HDM 735 4679.0000000000000000 9358 0 Magura MT7 3370.0000000000000000 25005.4 -202.2 Yeti Carbon 4299.9900000000000000 9008.48 -204.25 OC1 5000.0000000000000000 20000 0 Cannondale 2 Flat 4018.9900000000000000 25801.92 -281.33 Sram Guide RE 5899.0000000000000000 24716.8 -280.2 NS License Carb 5899.0000000000000000 6161.51 -262.51 NS License Carb 4449.9000000000000000 9398.18 -249.19 ProTaper AL 4299.0000000000000000 9189.12 -295.56 Evil Boomstick 7075 Alloy 3999.0000000000000000 8341.92 -171.96 Shimnano Deore M6100 4 5899.0000000000000000 42944.72 -235.96 NS License Carb 2199.0000000000000000 4700.36 -151.18 Sram G2 RSC 8000.0000000000000000 59360 -480 E13 TRS Base 1850.0000000000000000 1977.19 -127.19 Shimano Deore M6100 2 4700.0000000000000000 54802 -282 Newman Evo 5799.0000000000000000 12096.72 -249.36 Whyte 6061 Alloy 4854.9900000000000000 29129.94 0.00 OnOff Sulpher 1.0 3999.0000000000000000 8577.86 -289.93 Nukeproof Neutron V2 3999.0000000000000000 11997 0 Sram G2 R 3999.0000000000000000 8445.88 -223.94 Shimnano Deore M6100 4 4700.0000000000000000 10046.26 -323.13 Newman Evo 4350.0000000000000000 27144 -174 Nukeproof Horizon V2 6395.0000000000000000 6698.76 -303.76 RaceFace Turbine R 4700.0000000000000000 20116 -329 Newman Evo 2199.0000000000000000 9411.72 -153.93 Sram G2 RSC 7565.6600000000000000 15963.54 -416.11 Tektro HDM 735 4200.0000000000000000 40068 -252 Cannondale 3 Riser 4700.0000000000000000 9818.30 -209.15 Newman Evo 3999.9900000000000000 4171.99 -172.00 Intense Recon Alloy 4018.9900000000000000 8419.78 -190.90 Sram Guide RE 4399.0000000000000000 13955.82 -252.94 Specialized Trail 7050 Alloy 5800.0000000000000000 12412 -406 TranzX Burtted 6061 4854.9900000000000000 10127.50 -208.76 OnOff Sulpher 1.0 4499.0000000000000000 28613.64 -269.94 Ibis Carbon 3799.0000000000000000 26593 0 RaceFace Ride 5500.0000000000000000 34320 -220 OC3 Carbon 4499.0000000000000000 37431.68 -179.96 Ibis Carbon 1949.9900000000000000 2042.61 -92.62 RaceFace NextR Carbon 3673.9900000000000000 26746.65 -146.96 Sram Code R 3673.9900000000000000 7674.96 -163.49 Sram Code R 4199.0000000000000000 17467.84 -167.96 Ibis Aluminum 53

- What is one anomaly detected in the query results? some The price difference is  negative (that means a loss on every bike sale)
- What was the total sales costs for Rocky Mountain bikes? 324004.05

In [22]:
SELECT 
    c.comp_name AS Component_Name, 
    SUM(o.order_tot) AS Total_Sales 
FROM 
   dsci_504. Components c 
JOIN 
    dsci_504.Orders o 
ON 
    c.comp_id = o.prod_id 
WHERE 
    c.comp_name LIKE 'Rocky Mountain%' 
GROUP BY 
    c.comp_name;

(1 row(s) affected)

Total execution time: 00:00:00.015

component_name total_sales Rocky Mountain AM 324004.05

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">part 6: Discuss your confidence in using advanced techniques such as window functions after completing this assignment. What advantages do they provide over aggregate functions?</span>

<span style="color: rgb(45, 59, 69); font-family: LatoWeb, &quot;Lato Extended&quot;, Lato, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; background-color: rgb(255, 255, 255);">Answer:&nbsp;</span>  <span style="font-size: 16px;">the advantages of using window functions over aggregate functions include increased flexibility, the ability to work with more granular data, improved query performance, and more readable SQL code.&nbsp;</span>